tokio/src/task/join_set.rs

```diff
@@ -306,6 +306,31 @@ impl<T: 'static> JoinSet<T> {
         crate::future::poll_fn(|cx| self.poll_join_next_with_id(cx)).await
     }
 
+    /// Tries to join one of the tasks in the set that has completed and return its output.
+    ///
+    /// Returns `None` if the set is empty.    
+    pub fn try_join_next(&mut self) -> Option<Result<T, JoinError>> {
+        // Loop over all notified `JoinHandle`s to find one that's ready, or until none are left.
+        loop {
+            let mut entry = self.inner.try_pop_notified()?;
+
+            let res = entry.with_value_and_context(|jh, ctx| {
+                // Since this function is not async and cannot be forced to yield, we should
+                // disable budgeting when we want to check for the `JoinHandle` readiness.
+                let fut = unconstrained(jh);
+                pin!(fut);
+
+                fut.poll(ctx)
+            });
+
+            if let Poll::Ready(res) = res {
+                let _entry = entry.remove();
+
+                return Some(res);
+            }
```

Comment:
Please add a comment about why you go around the loop when it returns `Pending`.

Reply:
```rust 
// Loop over all notified `JoinHandle`s to find one that's ready, or until none are left.
```
I've added this comment at the top of the function to explain why the loop exists. Is it too concise?